src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -120,4 +125,19 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
 
     case s => violation(s"$s was not handled")
   }
+
+  /**
+    * Checks whether a struct whose type has name 'name' is cyclically defined (in terms of itself or other cyclic structures).
```

Comment:
`name` does not appear as any parameter. Also regarding the `visitedTypes`'s default value, it might make sense to change the parameters to be of type `PStructType` and `PIdnDef` and the recurse on some inner function

Reply:
I agree with your suggestions. I implemented them in a new commit